kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkConnector.java

```diff
@@ -86,4 +92,36 @@ public ConfigDef config() {
     return S3SinkConnectorConfig.getConfig();
   }
 
+  private boolean checkBucketExists(S3SinkConnectorConfig config) {
+    @SuppressWarnings("unchecked")
+    Class<? extends S3Storage> storageClass =
+            (Class<? extends S3Storage>)
+                    config.getClass(StorageCommonConfig.STORAGE_CLASS_CONFIG);
+
+    final S3Storage bucketCheckHelper = StorageFactory.createStorage(storageClass,
+            S3SinkConnectorConfig.class, config,
+            config.getString(StorageCommonConfig.STORE_URL_CONFIG));
+
+    return bucketCheckHelper.bucketExists();
+  }
+
+  @Override
+  public Config validate(final Map<String, String> connectorConfigs) {
+    ConfigDef configDef = this.config();
+    if (null == configDef) {
+      throw new ConnectException(String.format("%s.config() must return a ConfigDef that is not"
+              + " null.", this.getClass().getName()));
+    } else {
+
+      // Checking whether the bucket exists
+      if (!checkBucketExists(new S3SinkConnectorConfig(connectorConfigs))) {
```

Comment:
also if we do decide to have this, then refactor so that both connector and task uses the same method to create connection and verify for the existence of bucket

Reply:
validate() is called before the task is created; so we are trying to avoid creating a task if the bucket does not exist. Hence, we need it in the validate() method. 
We are already using a common method defined in the S3Storage class to verify bucket existence. We are just creating an object of S3Storage class using the configs and calling bucketExists() method in the class. The task has this object as a class variable.